week-1/calculator.rb

```diff
@@ -0,0 +1,30 @@
+puts "ingrese su operacion:"
+op = gets.chomp
+num = op.split(/(\d+\.?\d*)/).reject(&:empty?)
+i=0
+signos = ["+","-","*","/"]
+while i < num.length
+    if signos.include? (num[i])
+        case num[i]
+           when "**"
+                res=num[i-1].to_i.send :**,num[i+1].to_i 
+            when "/"
+                res=num[i-1].to_i.send :/,num[i+1].to_i
+            when "*"
+               res=num[i-1].to_i.send :*,num[i+1].to_i
+            when "+"
+                res=num[i-1].to_i.send :+,num[i+1].to_i
+            when "-"
+                res=num[i-1].to_i.send :-,num[i+1].to_i                       
+            else
+                res=0;
+        end
+        num[i-1]=res
+        num.delete_at(i)
+        num.delete_at(i)
```

Comment:
Línea duplicada

Reply:
tendriamos que eliminar el i y i+1 
pero al eliminar i automaticamente el elemento de indice i+1 se vuelve i 
esta duplicada por necesidad

Reply:
Quizá en vez de un case, mandarle directamente el signo!:
```ruby
sign = num[i]
number = num[i - 1].to_i
other_number = num[i + 1].to_i
res = number.send(sign.to_sym, other_number)
```

Reply:
(*english lesson about giving a code review in english) Why is line 23 duplicated? Can you give an explanation about it, please?